### PR TITLE
refactor!: remove builder return param from init

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2157

A separate `Init()` function seems kinda like an anti pattern. If it must always be called after `NewBuilder`, why doesn't `NewBuilder` call it? @mojtaba-esk

### CODE REVIEW

File: pkg/shares/share_builder.go [hunk 1]
```diff
@@ -32,18 +33,12 @@ func NewBuilder(ns appns.Namespace, shareVersion uint8, isFirstShare bool) *Buil
 	}
 }
 
-func (b *Builder) Init() (*Builder, error) {
+// Init initializes the share builder. It should be called right after NewBuilder.
+func (b *Builder) Init() error {
```
Comment (evan-forbes) [inline pkg/shares/share_builder.go:37]: apologies if I just forgot, but if this function should be called right after `NewBuilder` then why don't we just call it there?

**Review by evan-forbes — APPROVED** [review]

**Comment by mojtaba-esk** [comment]
> A separate `Init()` function seems kinda like an anti pattern. If it must always be called after `NewBuilder`, why doesn't `NewBuilder` call it? @mojtaba-esk

Don't remember the rational behind it, I checked the code and probably the reason does not exist any more. So we can merge the two functions.

@evan-forbes ^

**Review by mojtaba-esk — CHANGES_REQUESTED** [review]
LGTM, please go on with the proposal to merge the Init into the NewBuilder

File: pkg/shares/share_builder.go (hunk 1 above)
Comment (rootulp) [reply pkg/shares/share_builder.go]: Resolved via https://github.com/celestiaorg/celestia-app/pull/2165/commits/cb86e485f193c363038fc598ded216c0f1863350

File: pkg/shares/share_builder.go [hunk 2]
```diff
@@ -22,28 +22,27 @@ func NewEmptyBuilder() *Builder {
...
-	return &Builder{
+// NewBuilder returns a new share builder.
+func NewBuilder(ns appns.Namespace, shareVersion uint8, isFirstShare bool) (*Builder, error) {
+	b := Builder{
 		namespace:      ns,
 		shareVersion:   shareVersion,
 		isFirstShare:   isFirstShare,
 		isCompactShare: isCompactShare(ns),
 	}
+	err := b.init()
```
Comment (mojtaba-esk) [inline pkg/shares/share_builder.go:33]: looks good, could even be more concise by having the init in the if phrase like this:

```
if err := b.init(); err != nil {
    return nil, err
}
```

File: pkg/shares/share_builder_test.go [hunk 3]
```diff
@@ -318,3 +311,10 @@ func TestShareBuilderImportRawData(t *testing.T) {
 		})
 	}
 }
+
+// mustNewBuilder returns a new builder with the given parameters. It fails the test if an error is encountered.
+func mustNewBuilder(t *testing.T, ns appns.Namespace, shareVersion uint8, isFirstShare bool) *Builder {
```
Comment (mojtaba-esk) [inline pkg/shares/share_builder_test.go:316]: smart idea

**Review by mojtaba-esk — DISMISSED** [review]

File: pkg/shares/share_builder.go (hunk 2 above)
Comment (rootulp) [reply pkg/shares/share_builder.go]: 109bff6aeee94f5a7bf7ab1b120343ea370ae945

**Review by mojtaba-esk — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]
Yeah this makes sense to me

**Pull request merged**